subsys/bluetooth/scan.c

```diff
@@ -283,6 +319,141 @@ static void notify_connecting_error(struct bt_scan_device_info *device_info)
 	}
 }
 
+static bool blacklist_device_check(const bt_addr_le_t *addr)
+{
+	bool blacklist_device = false;
+
+	k_mutex_lock(&blacklist_mutex, K_FOREVER);
+
+	for (size_t i = 0; i < bt_scan.blacklist.cnt; i++) {
+		if (bt_addr_le_cmp(&bt_scan.blacklist.addr[i], addr) == 0) {
+			blacklist_device = true;
+		}
+	}
+
+	k_mutex_unlock(&blacklist_mutex);
+
+	return blacklist_device;
+}
+
+static void attempts_filter_force_add(struct conn_attempts_filter *filter,
+				      const bt_addr_le_t *addr)
+{
+	/* Overwrite the oldest device */
+	filter->oldest_device->attempts = 0;
+	bt_addr_le_copy(&filter->oldest_device->addr, addr);
+
+	if (filter->oldest_device == &filter->device[filter->count - 1]) {
```

Comment:
that would be problematic if `filter->count` is zero

Reply:
`ARRAY_SIZE(filter->device)` maybe?

Reply:
Fixed